backend/src/plugin/plugin_manager.cpp

```
@@ -3,19 +3,36 @@
 
 #include "plugin_manager.hpp"
 #include "descriptors/engine_config_descriptor.hpp"
+#include "descriptors/engine_descriptor.hpp"
 #include "fake_plugin.hpp"
 #include "hipdnn_exception.hpp"
 
 namespace hipdnn_backend
 {
 
-void Plugin_manager::initialize(
-    /* some stuff to help you find which plugins should be loaded, but for now blank */)
+void Plugin_manager::initialize(/* heuristics */)
 {
+
     // TODO : actually find and init the plugins properly
     // for now we will just use a fake plugin.
-    // # DISCUSS: how do we find the plugins?
-    _plugins.push_back(std::make_shared<Fake_plugin>());
+    // # DISCUSS: how do we find the plugins? store in json?
+    auto fake_plugin = std::make_shared<Fake_plugin>();
+
+    // for all the applicable engines, add the same plugin to the map
+    for(const int64_t& engine_id : fake_plugin->get_applicable_engines(nullptr))
+    {
+        _plugins.insert({engine_id, fake_plugin});
+    }
+}
+
+std::shared_ptr<Hipdnn_plugin_base> Plugin_manager::get_plugin(int64_t engine_id)
+{
+    auto plugin_iter = _plugins.find(engine_id);
+    if(plugin_iter == _plugins.end())
+    {
+        return nullptr;
```

Should throw if it's not found.